node/src/types/validator_matrix.rs

```diff
@@ -279,3 +292,78 @@ impl EraValidatorWeights {
         SignatureWeight::Insufficient
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::iter;
+
+    use casper_types::EraId;
+    use num_rational::Ratio;
+
+    use crate::{
+        components::consensus::tests::utils::{ALICE_PUBLIC_KEY, ALICE_SECRET_KEY},
+        types::validator_matrix::MAX_VALIDATOR_MATRIX_ENTRIES,
+    };
+
+    use super::{EraValidatorWeights, ValidatorMatrix};
+
+    fn empty_era_validator_weights(era_id: EraId) -> EraValidatorWeights {
+        EraValidatorWeights::new(
+            era_id,
+            iter::once((ALICE_PUBLIC_KEY.clone(), 100.into())).collect(),
+            Ratio::new(1, 3),
+        )
+    }
+
+    #[test]
+    fn register_validator_weights_pruning() {
+        // Create a validator matrix and saturate it with entries.
+        let mut validator_matrix = ValidatorMatrix::new_with_validator(ALICE_SECRET_KEY.clone());
+        let mut era_validator_weights = vec![validator_matrix.validator_weights(0.into()).unwrap()];
+        era_validator_weights.extend(
+            (1..MAX_VALIDATOR_MATRIX_ENTRIES as u64)
+                .into_iter()
+                .map(EraId::from)
+                .map(empty_era_validator_weights),
+        );
+        for evw in era_validator_weights
+            .iter()
+            .take(MAX_VALIDATOR_MATRIX_ENTRIES)
+            .skip(1)
+            .cloned()
+        {
+            assert!(validator_matrix.register_era_validator_weights(evw));
+        }
+
+        // Now that we have 6 entries in the validator matrix, try adding more.
+        // We should have an entry for era 3 (we have eras 0 through 5
+        // inclusive).
+        assert!(validator_matrix.has_era(&(MAX_VALIDATOR_MATRIX_ENTRIES as u64 / 2).into()));
+        // Add era 7.
+        era_validator_weights.push(empty_era_validator_weights(
+            (MAX_VALIDATOR_MATRIX_ENTRIES as u64 + 1).into(),
+        ));
+        // Now the entry for era 3 should be dropped and we should be left with
+        // the 3 lowest eras [0, 1, 2] and 3 highest eras [4, 5, 6].
```

Comment:
I'd expect to have [0, 1, 2, 4, 5, **7**] here.

Or did you mean to add Era 6 [here](https://github.com/casper-network/casper-node/pull/3528/files#diff-fc4d1befdc75c72b75fe10cb728dc49dd320576ab84a56971e007667599c2cacR344).

Reply:
I meant to add the 7th entry, so that would be era `6` if it was continuous, could be era `7` or any greater number for the purpose of the test. To keep it consistent, I'll make it era `6`, which would be the 7th entry.